tests/unit/common/test_storage.py

```diff
@@ -101,11 +104,12 @@ def test_caching_storage(tmpdir: py.path.local, mocker: MockerFixture) -> None:
     assert storage.list_jsons() == [TEST_JSON]
 
     # We shouldn't wind up in download method of rohmu at all.
-    mockdown = mocker.patch.object(storage.backend_storage, "download_json")
+    mockdown = mocker.patch.object(storage.backend_storage, "open_json_bytes")
     # Nor list hexdigests
     mocklist = mocker.patch.object(storage.backend_storage, "list_jsons")
 
-    assert storage.download_json(TEST_JSON) == TEST_JSON_DATA
+    with storage.open_json_bytes(TEST_JSON) as json_bytes:
+        assert json.loads(bytes(json_bytes)) == TEST_JSON_DATA
```

Comment:
This is a bit sad that we are faking types like this that can have confusing errors.

Reply:
Where are we faking types ? (the call to bytes is not a C cast, it's a real conversion to a bytes object)

Reply:
mypy thinks `json_bytes` here is a bytearray, which `json.loads` accepts.  But in fact it is a mmap object and `json.loads` fails with a `TypeError`.

Reply:
Aah, you're right that's a bad typing issue. I didn't realize I did that, I'll adjust the types :+1: 

(That might mean I'll have to expose the mmap type everywhere, but I'd rather have truthful code even if the truth isn't pretty.)

Reply:
Yea agreed I would go with the truth even if it is ugly